backend/src/main/java/com/woowacourse/kkogkkog/coupon/application/CouponService.java

```diff
@@ -46,8 +47,12 @@ public CouponService(MemberRepository memberRepository,
     }
 
     @Transactional(readOnly = true)
-    public CouponDetailResponse find(Long couponId) {
+    public CouponDetailResponse find(Long memberId, Long couponId) {
+        Member member = findMember(memberId);
         Coupon coupon = findCoupon(couponId);
+        if (!coupon.isSenderOrReceiver(member)) {
+            throw new CouponNotAccessibleException();
+        }
         List<CouponHistory> couponHistories = couponHistoryRepository.findAllByCouponIdOrderByCreatedTimeDesc(
             couponId);
```

Comment:
의견 묻고 싶은데요, 이런 것들도 전부 변수추론을 하는건 어떠할까요?

```suggestion
        final var couponHistories = couponHistoryRepository.findAllByCouponIdOrderByCreatedTimeDesc(couponId);
```